lib/server/services/run/updateRun.js

```diff
@@ -111,6 +112,11 @@ exports.updateRun = async (identifier, payload, transaction) => {
         // Send notification if run quality has changed
         if (previousRun.runQuality !== runModel.runQuality) {
             const { runQualityChangeReason } = metadata;
+
+            const { lhcPeriodId } = runModel;
+            const lhcPeriod = await getOrCreateLhcPeriod(lhcPeriodId);
```

Comment:
As you do not have any information on the name of the LHC Period, it is not a good idea here to create an lhc period just with the ID, as the ID is auto-incremented in the DB.

I would recommend to add a similar service for retrieving or failing the lhc period by id. (`getLhcPeriodOrFail.js`. You already have some examples getLogOrFail, etc.). Moreover, you already have the method for `findOne` in the Repository interface.

Reply:
Moreover, if the period name cannot be found, then the user should not be impacted and the text parts for log entry for LHC Period should simply specify `UNKNOWN` for the lhc period